src/features/ExtensionCommands.ts

```diff
@@ -380,24 +385,34 @@ export class ExtensionCommandsFeature implements IFeature {
     private closeFile(filePath: string): Thenable<EditorOperationResponse> {
 
         var promise: Thenable<EditorOperationResponse>;
-
-        // Make sure the file path is absolute
-        if (!path.win32.isAbsolute(filePath))
+        if (this.findTextDocument(this.normalizeFilePath(filePath)))
         {
-            filePath = path.win32.resolve(
-                vscode.workspace.rootPath,
-                filePath);
+            promise =
+                vscode.workspace.openTextDocument(filePath)
+                    .then(doc => vscode.window.showTextDocument(doc))
+                    .then(editor => vscode.commands.executeCommand("workbench.action.closeActiveEditor"))
+                    .then(_ => EditorOperationResponse.Completed);
+        }
+        else
+        {
+            promise = Promise.resolve(EditorOperationResponse.Completed);
         }
 
-        // Normalize file path case for comparison
-        var normalizedFilePath = filePath.toLowerCase();
+        return promise;
+    }
+
+    private saveFile(filePath: string): Thenable<EditorOperationResponse> {
 
-        if (vscode.workspace.textDocuments.find(doc => doc.fileName.toLowerCase() == normalizedFilePath))
+        var promise: Thenable<EditorOperationResponse>;
```

Comment:
Another linter thing and something @daviwil espoused as well - favor the use of  `let` over `var`.  `let` is properly blocked scoped in JavaScript.

Reply:
Fixed all occurrences in the file :)